lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -597,16 +601,32 @@ impl<'s> SegmentHolder {
                     self.aloha_random_write(
                         &appendable_segments,
                         |_appendable_idx, appendable_write_segment| {
-                            let all_vectors = write_segment.all_vectors(point_id)?;
-                            let payload = write_segment.payload(point_id)?;
+                            let mut all_vectors = write_segment.all_vectors(point_id)?;
+                            let mut payload = write_segment.payload(point_id)?;
 
-                            appendable_write_segment.upsert_point(op_num, point_id, all_vectors)?;
-                            appendable_write_segment
-                                .set_full_payload(op_num, point_id, &payload)?;
+                            let apply_payload =
+                                point_cow_operation(point_id, &mut all_vectors, &mut payload)?;
+
+                            let mut applied = true;
+
+                            applied &= appendable_write_segment.upsert_point(
+                                op_num,
+                                point_id,
+                                all_vectors,
+                            )?;
```

Comment:
Let's simplify this to:

```suggestion
                            let applied = appendable_write_segment.upsert_point(
                                op_num,
                                point_id,
                                all_vectors,
                            )?;
```